climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1158 @@
+        hist_fire_seasons = [] # list to save fire seasons
+
+        for year in years:
+            LOGGER.info('Setting up historical fire seasons %s.', str(year))
+            firms_temp = self._select_fire_season(firms, year, hemisphere=hemisphere)
+            # calculate historic fire seasons
+            wf_year = WildFire()
+            wf_year.set_hist_fire_FIRMS(firms_temp, centroids=centroids)
+            hist_fire_seasons.append(wf_year)
+
+        # fires season (used to define distribution of n fire for the
+        # probabilistic fire seasons)
+        n_fires = np.zeros(len(years))
+        for i, wf in enumerate(hist_fire_seasons):
+            n_fires[i] = len(wf.event_id)
+
+        if keep_all_fires:
+            self.hist_fire_seasons = hist_fire_seasons
+
+        # save
+        self.tag = TagHazard('WFseason')
+        self.centroids = centroids
+        self.n_fires = n_fires
+        self.units = 'K' # Kelvin brightness
+
+        # Following values are defined for each fire
+        self.event_id = np.arange(1, len(years)+1).astype(int)
+        self.event_name = list(map(str, years))
+        self.date = np.zeros(len(years), int)
+        self.date_end = np.zeros(len(years), int)
+        if hemisphere == 'NHS':
+            for y_idx, year in enumerate(years):
+                self.date[y_idx] = date.toordinal(date(year, 1, 1))
+                self.date_end[y_idx] = date.toordinal(date(year, 12, 31))
+        elif hemisphere == 'SHS':
+            for y_idx, year in enumerate(years):
+                self.date[y_idx] = date.toordinal(date(year, 7, 1))
+                self.date_end[y_idx] = date.toordinal(date(year+1, 6, 30))
+        self.orig = np.ones(len(years), bool)
+        self._set_frequency()
+
+        # Following values are defined for each fire and centroid
+        self.intensity = sparse.lil_matrix(np.zeros((len(years), len(centroids.lat))))
+        for idx, wf in enumerate(hist_fire_seasons):
+            self.intensity[idx] = wf.intensity.max(axis=0)
+        self.intensity = self.intensity.tocsr()
+        self.fraction = self.intensity.copy()
+        self.fraction.data.fill(1.0)
+
+    def set_proba_fire_seasons(self, n_fire_seasons=1, n_ignitions=None,
+                               keep_all_fires=False):
+        """ Generate probabilistic fire seasons. Fire seasons
+        are created by running n probabilistic fires per year which are then
+        summarized into a probabilistic fire season by calculating the max
+        intensity at each centroid for each probabilistic fire season.
+        Probabilistic fires are created using the logic described in the
+        method 'run_one_bushfire'.
+
+        The fire propagation matrix can be assigned separately, if that is not
+        done it will be generated on the available historic fire (seasons).
+
+        Intensities are drawn randomly from historic events. Thus, this method
+        requires at least one fire to draw from.
```

Comment:
This requires a bit more explanation. Either in a future publication, or better already here in the docstring.